cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_csr_impl.hpp

```diff
@@ -162,6 +165,9 @@ sycl::event assign_clusters(sycl::queue& q,
                                Float(0),
                                { transpose_event });
 
+    // Workaround to sparse K-means abort on PVC.
+    dist_event.wait();
```

Comment:
Are you sure that these waits are necessary? Because looks like at least select function contains inside cgh.depends_on checker

Reply:
agreed - seems like cpp/oneapi/dal/algo/kmeans/backend/gpu/train_kernel_lloyd_csr_dpc.cpp had a true sycl event problem whereas these might not be necessary

Reply:
@Alexandr-Solovev , @ethanglaser 
I understand that those two waits look super-weird, but without them the problem remains. I can remove and we'll see what happen in CI.